backend/pkg/server/api/handler.go

```diff
@@ -108,76 +111,95 @@ func (h *Handler) GetResourceDetails(ctx *gin.Context) {
 func (h *Handler) GetCustomBoard(ctx *gin.Context) {
 	var err error
 
-	config, ok := h.boards[ctx.Param("id")]
-	if !ok {
-		ctx.AbortWithStatus(http.StatusNotFound)
-		return
-	}
+	session := sessions.Default(ctx)
 
-	endpoints, ok := h.clients[ctx.Param("cluster")]
-	if !ok {
-		ctx.AbortWithStatus(http.StatusNotFound)
-		return
-	}
+    // Get the profile from the session
+    profile := session.Get("profile")
```

Comment:
You can use `auth.ProfileFrom(ctx)` which returns `nil` if no profile is available